src/monodroid/jni/new_delete.cc

```diff
@@ -0,0 +1,76 @@
+#include <stdlib.h>
+#include <new>
+
+extern "C" {
+#include "java-interop-util.h"
+}
+
+void *
+operator new (size_t size)
+{
+    if (size == 0)
+        size = 1;
+
+    void* p = ::malloc (size);
+    if (p == nullptr) {
+	    log_fatal (LOG_DEFAULT, "Out of memory in the `new` operator");
+	    exit (FATAL_EXIT_OUT_OF_MEMORY);
+    }
+
+    return p;
+}
+
+void*
+operator new (size_t size, const std::nothrow_t&) noexcept
+{
+    return ::operator new(size);
+}
+
+void*
+operator new[] (size_t size)
+{
+    return ::operator new(size);
+}
+
+void*
+operator new[] (size_t size, const std::nothrow_t&) noexcept
+{
+	return ::operator new[](size);
+}
+
+void
+operator delete (void* ptr) noexcept
```

Comment:
It's truly a pity that the `delete` operator doesn't take a `void*&` parameter, so it could *null out the parameter automatically*...